File: c/leetcode/206. Reverse Linked List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

struct ListNode *reverseList(struct ListNode *head)
{
  struct ListNode *prev = NULL;
  struct ListNode *current = NULL;

  while (head)
  {
    current = head;
    head = head->next;
    current->next = prev;
    prev = current;
  }

  return current;
}

File: go/leetcode/206. Reverse Linked List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
    var prev *ListNode
    var current *ListNode
    
    for head != nil {
        current = head;
        head = head.Next;
        current.Next = prev;
        prev = current;
    }
    
    return current;
}

 